src/wallet/wallet.ts

```diff
@@ -302,13 +315,59 @@ class HathorWalletServiceWallet extends EventEmitter implements IHathorWallet {
    * @memberof HathorWalletServiceWallet
    * @inner
    */
-  async start({ pinCode, password }) {
+  async start({ pinCode, password }: { pinCode?: string, password?: string } = {}) {
     if (!pinCode) {
       throw new Error('Pin code is required when starting the wallet.');
     }
 
     this.setState(walletState.LOADING);
 
+    let hasAccessData: boolean;
+    try {
+      const accessData = await this.storage.getAccessData();
+      hasAccessData = !!accessData;
+    } catch (err) {
+      if (err instanceof UninitializedWalletError) {
+        hasAccessData = false;
+      } else {
+        throw err;
+      }
+    }
+    if (!hasAccessData) {
+      let accessData: IWalletAccessData;
+      if (this.seed) {
+        if (!password) {
+          throw new Error('Password is required when starting the wallet from the seed.');
+        }
+        accessData = walletUtils.generateAccessDataFromSeed(
+          this.seed,
+          {
+            passphrase: this.passphrase,
+            pin: pinCode,
+            password,
+            networkName: this.network.name,
+            // multisig: not implemented on wallet service yet
+          }
+        );
+      } else if (this.xpriv) {
+        // generateAccessDataFromXpriv expects a xpriv on the change level path
+        const accountLevelPrivKey = new bitcore.HDPrivateKey(this.xpriv);
+        const changeLevelPrivKey = accountLevelPrivKey.deriveNonCompliantChild(0);
+
+        accessData = walletUtils.generateAccessDataFromXpriv(
+          changeLevelPrivKey.xprivkey,
+          {
+            pin: pinCode,
+            // multisig: not implemented on wallet service yet
+          },
+        );
+        accessData.authKey = encryptData(this.authPrivKey.xprivkey!, pinCode);
```

Comment:
Why don't we move this line to the `generateAccessDataFromXpriv` method? We already handle this for other accessData keys there, right?

Reply:
Moved to the wallet utils as requested